sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,12 +346,25 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi
```

Comment:
I imported it there itself because not every method needs it (also I saw some imports done this way in the code). Does it make a big difference though?